lib/dicom_net/association.ex

```diff
@@ -87,16 +87,18 @@ defmodule DicomNet.Association do
     %{associate_request | presentation_contexts: accepted_presentation_contexts}
   end
 
-  defp handle_pdu(
-         %Pdu{
-           type: :associate_request,
-           data: associate_request
-         },
-         %{socket: _socket, state: :waiting_for_association} = state
+  defp reject_associate_request(
+         %{presentation_contexts: presentation_contexts} = associate_request
        ) do
-    Logger.info("Accepting association")
-    association_data = accept_associate_request(associate_request)
+    accepted_presentation_contexts =
+      presentation_contexts
+      |> Enum.map(fn {id, pc} -> {id, Map.put(pc, :result, :reject)} end)
+      |> Enum.into(%{})
```

Comment:
1. I guess this also somehow works for prohibiting a communication, but DICOM uses a separate response PDU for rejecting associations which contains e.g. fields for the reason. See [here](https://dicom.nema.org/dicom/2013/output/chtml/part08/sect_9.3.html#sect_9.3.4).
2. That aside, if you copy/paste a whole function to just replace once value inside, probably making that value a parameter of the function is the better option ;)